README.adoc

```diff
@@ -33,26 +33,18 @@ To query the properties applicable to a file in a source tree:
 
 [source,java]
 ----
-Cache myCache = ...; // e.g. Caches.none()
-EditorConfigLoader myLoader = ...; // e.g. EditorConfigLoader.getDefault()
-EditorConfigSession mySession = EditorConfigSession.builder()
+Cache myCache = ...; // e.g. Caches.permanent()
+EditorConfigLoader myLoader = ...; // e.g. EditorConfigLoader.default_()
+ResourcePropertiesService propService = ResourcePropertiesService.builder()
```

Comment:
Why do you wish to rename to ResourcePropertiesService ? I liked the EditorConfigSession name because it contained "EditorConfig"

Reply:
Yes, renamings definitely need a justification. The justification in this case is that `ResourcePropertiesService` describes better what the class does than `EditorConfigSession`. 

When I originally proposed `EditorConfigSession`, I have seen it as a an analogon to some kind of a server environment where there are sessions that entail several requests per user. However that meaning of "session" does not quite suit our situation, because (1) our code has no notion of a user and (2) what we do in `queryProperties()` is *serving* some requests and sessions do not serve; it is rather services that serve. By "serve" I mean performing some sort of a business logic, what is definitely the case in `queryProperties()`. Sessions can contain some state, but I do not think they should perform business logic. That's for the switch from `*Session` to `*Service`.

Now for `EditorConfig*` -> `ResourceProperties*`: We have an `EditorConfig` class in our model and we also now have `ResourceProperties`. Because what `queryProperties()` returns is `ResourceProperties` rather than `EditorConfig`, I think that `ResourcePropertiesService` is a better name for the class. It makes a lot of sense to say that `ResourcePropertiesService` serves  `ResourceProperties`.

WDYT?

Reply:
Nice explanation! I love it:) Waiting for a PR for ec4e to accept this PR when you think you have finished. Thanks!